### PR TITLE
internal/gengapic: use gtransport.DialPool, deprecate Connection()

### PR DESCRIPTION
**Pull request by broady**
Connection pooling via gtransport.Dial will be removed soon.

See https://github.com/googleapis/google-api-go-client/issues/441

### CODE REVIEW

**Comment by broady** [comment]
I'm finding it very difficult to update the go.mod and go.sum files in the showcase directory.

```
gapic-generator-go/showcase +dialpool $ go get google.golang.org/api/transport@latest && go mod tidy
go: cloud.google.com/go/showcase@v0.0.0: parsing gen/cloud.google.com/go/showcase/go.mod: open /usr/local/google/home/cbro/modsrc/gapic-generator-go/showcase/gen/cloud.google.com/go/showcase/go.mod: no such file or directory

< comment out replace directive >

gapic-generator-go/showcase +dialpool $ go get google.golang.org/api/transport@latest && go mod tidy
go: cloud.google.com/go/showcase@v0.0.0: unrecognized import path "cloud.google.com/go/showcase" (parse https://cloud.google.com/go/showcase?go-get=1: no go-import meta tags ())

< comment out require entry for showcase in go.mod >

gapic-generator-go/showcase +dialpool $ go get google.golang.org/api/transport@latest && go mod tidy
go: downloading cloud.google.com/go v0.45.0
go: extracting cloud.google.com/go v0.45.0
go: downloading github.com/googleapis/gapic-showcase v0.5.0
go: extracting github.com/googleapis/gapic-showcase v0.5.0
showcase_integration tested by
        showcase_integration.test imports
        cloud.google.com/go/showcase/apiv1beta1: module cloud.google.com/go@latest found (v0.52.0), but does not contain package cloud.google.com/go/showcase/apiv1beta1
```

I've just made the change to go.mod manually.

File: internal/gengapic/client_init.go [hunk 1]
```diff
@@ -207,8 +207,8 @@ func (g *generator) clientInit(serv *descriptor.ServiceDescriptorProto, servName
 		p("// Methods, except Close, may be called concurrently. However, fields must not be modified concurrently with method calls.")
 		p("type %sClient struct {", servName)
 
-		p("// The connection to the service.")
-		p("conn *grpc.ClientConn")
+		p("// Connection pool of gRPC connections to the service.")
+		p("connp gtransport.ConnPool")
```
Comment (tbpg) [inline internal/gengapic/client_init.go]: Totally minor, but I think I prefer `connPool`. Not clear what `connp` stands for. Feel free to ignore.

File: internal/gengapic/client_init.go [hunk 2]
```diff
@@ -247,12 +247,12 @@ func (g *generator) clientInit(serv *descriptor.ServiceDescriptorProto, servName
...
+		p("  connp, err := gtransport.DialPool(ctx, append(default%sClientOptions(), opts...)...)", servName)
 		p("  if err != nil {")
 		p("    return nil, err")
 		p("  }")
 		p("  c := &%sClient{", servName)
-		p("    conn:        conn,")
+		p("    connp:       connp,")
 		p("    CallOptions: default%sCallOptions(),", servName)
 		p("")
 		p("    %s: %s.New%sClient(conn),", grpcClientField(servName), imp.Name, serv.GetName())
```
Comment (tbpg) [inline internal/gengapic/client_init.go]: Is it correct to use `conn` here?

Comment (broady) [reply internal/gengapic/client_init.go]: absolutely it is incorrect

File: internal/gengapic/client_init.go (hunk 1 above)
Comment (broady) [reply internal/gengapic/client_init.go]: done

**Comment by noahdietz** [comment]
For the `showcase/go.mod`, it is just meant as an integration test. Running `make test` starts a showcase binary, generates a client for the API and runs the tests for them that exist in that directory. That might also update the go.mod/go.sum. A manual change is fine too.

**Comment by noahdietz** [comment]
Fixes #46 too.

**Comment by broady** [comment]
Hurrah!

Fixing merge conflict.

**Pull request merged**